### PR TITLE
ADCS Commander

### PR DESCRIPTION
**Pull request by shihaocao**
This PR brings in the ADCSCommander which creates all the command state fields that ADCSController responds to.

It also adds the ADCS functional fault to ADCSBoxMonitor that is signaled() if the ADCSC stops responding to is_functional() calls.

It also implements the doubling up on HAVT command tables within ADCSController.

Note: Commander currently does not actually compute the correct wheel torques and mtr torques to send to the ADCSController. That will come in the next ADCS related PR.

### CODE REVIEW

File: test/test_fsw_adcs_commander/test_adcs_commander.cpp [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        std::vector<WritableStateField<bool>*> havt_cmd_reset_vector_fp;
+        std::vector<WritableStateField<bool>*> havt_cmd_disable_vector_fp;
+
+        /**
+         * @brief If this statefield is true, then the cmd_table is uploaded to ADCS
+         * 
+         * Statefield is set to true for autonomous HAVT response, or by ground command
+         */
+        // TODO PROBAABLY DELETE
+        WritableStateField<bool>* havt_cmd_apply_f;
```
Comment (tanishqaggarwal) [inline test/test_fsw_adcs_commander/test_adcs_commander.cpp]: Let's resolve this todo

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 2]
```diff
@@ -30,6 +30,8 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
     gyr_vec_flag("adcs_monitor.gyr_vec_flag", flag_sr),
     gyr_temp_flag("adcs_monitor.gyr_temp_flag", flag_sr),
     havt_bool_sr(),
+    adcs_is_functional("adcs_monitor.functional", flag_sr),
+    adcs_functional_fault("adcs_monitor.functional_fault", 1, control_cycle_count),
```
[the remark this answers is not in the document]
Comment (shihaocao) [reply src/fsw/FCCode/ADCSBoxMonitor.cpp]: In the same vein of logic of having the read HAVT in addition to the read HAVT throw flags (for wheels), I think it's useful to have a state field that represents the actual return of the is_functional() call. The bool state field within a Fault will only return whether or not it is faulted even if you call the ->get() directly.

File: src/fsw/FCCode/MainControlLoop.hpp [hunk 3]
```diff
@@ -82,7 +84,8 @@ class MainControlLoop : public ControlTask<void> {
         static constexpr unsigned int uplink_consumer_offset     =  61500;
         static constexpr unsigned int mission_manager_offset     =  61600;
         static constexpr unsigned int attitude_computer_offset   =  61700;
-        static constexpr unsigned int adcs_box_controller_offset =  97400;
+        static constexpr unsigned int adcs_commander_offset      =  97400;
+        static constexpr unsigned int adcs_box_controller_offset =  97900;
```
[the remark this answers is not in the document]
Comment (shihaocao) [reply src/fsw/FCCode/MainControlLoop.hpp]: Done

File: src/fsw/FCCode/MainControlLoop.cpp [hunk 4]
```diff
@@ -87,7 +88,8 @@ void MainControlLoop::execute() {
     attitude_estimator.execute_on_time();
     mission_manager.execute_on_time();
     attitude_computer.execute_on_time();
-    // adcs_box_controller.execute_on_time(); // Commented out since commander is needed for full functionality
+    adcs_commander.execute_on_time();
+    adcs_box_controller.execute_on_time();
```
[the remark this answers is not in the document]
Comment (shihaocao) [reply src/fsw/FCCode/MainControlLoop.cpp]: Right. Will do

File: src/fsw/FCCode/ADCSBoxMonitor.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (tanishqaggarwal) [reply src/fsw/FCCode/ADCSBoxMonitor.cpp]: Fair.

File: test/test_fsw_mission_manager/test_mission_manager.cpp [hunk 5]
```diff
@@ -332,11 +334,12 @@ void test_adcs_faults_inithold() {
     // hold if the satellite is in startup past its
     // deployment wait period.
     {
-        std::array<TestFixture, 4> tf_v;
+        std::array<TestFixture, 5> tf_v;
         tf_v[0].wheel1_adc_fault_f.override();
```
Comment (tanishqaggarwal) [inline test/test_fsw_mission_manager/test_mission_manager.cpp]: You didn't actually change the fault here

File: src/fsw/FCCode/ADCSBoxMonitor.hpp [hunk 6]
```diff
@@ -86,7 +86,11 @@ class ADCSBoxMonitor : public TimedControlTask<void>
     std::vector<ReadableStateField<bool>> havt_read_vector;
     Serializer<bool> havt_bool_sr;
 
-    // Fault is signaled based off of HAVT
+    //! Fault is thrown if is_functional() for ADCS itself doesn't work
+    ReadableStateField<bool> adcs_is_functional;
```
Comment (nhz2) [inline src/fsw/FCCode/ADCSBoxMonitor.hpp]: Could you add the string name of the state field to the header documentation, just so it is easier to find,(unless its already documented somewhere else).

File: src/fsw/FCCode/ADCSCommander.hpp [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    ADCSCommander(StateFieldRegistry& registry, unsigned int offset);
+
+    /**
+     * @brief Determine control outputs for the ADCSBoxController
+     */
+    void execute() override;
+
+   protected:
+    // input fields
+    const WritableStateField<unsigned char>* adcs_state_fp;
```
Comment (nhz2) [inline src/fsw/FCCode/ADCSCommander.hpp]: What enum type is this? Add that to the documentation.

File: src/fsw/FCCode/MainControlLoop.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (tanishqaggarwal) [reply src/fsw/FCCode/MainControlLoop.cpp]: We discussed it's OK to merge, unless you have time to try it before @kkrol27 approves

**Pull request merged**